### PR TITLE
op-node: specify Delta upgrade version, update superchain-registry, bump protocol-support to Delta pre-release

### PR DESCRIPTION
**Pull request by protolambda**
**Description**

Changes:
- Superchain upgrades spec:
  - Link governance proposal to Canyon version entry
  - Add Delta version entry
  - Add Delta upgrade description: span-batches feature, timestamp based activation
  - Add placeholder definitions for Eclipse and Fjord
- Pull in superchain-registry update. This covers:
  - Delta devnet hardfork time - 8 days from now!
  - Definition of Eclipse and Fjord hardfork configuration variables.
- Add Fjord and Eclipse to the deploy-config
- Add Fjord and Eclipse to the rollup config
- Bump op-node protocol-version support declaration to 5.0.0-1, i.e. the Delta pre-release for the devnet.

## Summary by CodeRabbit

- **New Features**
  - Introduced new configuration parameters for network upgrades, enhancing the deployment process with time offset calculations.
  - Added activation time settings for the Eclipse and Fjord network upgrades.

- **Documentation**
  - Updated the superchain upgrades documentation to reflect new network upgrade stages and governance proposals.

- **Refactor**
  - Expanded system configuration to incorporate new time-related fields for network upgrade activation checks.

- **Style**
  - Improved method descriptions to include information about network upgrade activation times.

### CODE REVIEW

File: op-chain-ops/genesis/config.go [hunk 1]
```diff
@@ -115,6 +115,12 @@ type DeployConfig struct {
 	// L2GenesisDeltaTimeOffset is the number of seconds after genesis block that Delta hard fork activates.
 	// Set it to 0 to activate at genesis. Nil to disable Delta.
 	L2GenesisDeltaTimeOffset *hexutil.Uint64 `json:"l2GenesisDeltaTimeOffset,omitempty"`
+	// L2GenesisDeltaTimeOffset is the number of seconds after genesis block that Eclipse hard fork activates.
```
[the remark this answers is not in the document]
Comment (protolambda) [reply op-chain-ops/genesis/config.go]: Thank you little rabbit

**Review by ajsutton — APPROVED** [review]
LGTM other than where the excessively verbose little rabbit had a couple of good points. :)

File: op-node/rollup/superchain.go [hunk 2]
```diff
@@ -12,7 +12,7 @@ import (
 	"github.com/ethereum-optimism/superchain-registry/superchain"
 )
 
-var OPStackSupport = params.ProtocolVersionV0{Build: [8]byte{}, Major: 4, Minor: 0, Patch: 0, PreRelease: 1}.Encode()
+var OPStackSupport = params.ProtocolVersionV0{Build: [8]byte{}, Major: 5, Minor: 0, Patch: 0, PreRelease: 1}.Encode()
```
Comment (trianglesphere) [inline op-node/rollup/superchain.go:15]: We're gonna run into the issue where we may never cut a version at v4.0.0

**Review by trianglesphere — APPROVED** [review]

File: op-node/rollup/superchain.go (hunk 2 above)
Comment (protolambda) [reply op-node/rollup/superchain.go:15]: The version is meant to indicate the protocol it supports up to. There doesn't strictly need to be a whole release in this case. 5.0.0-1 implies 4.x.x support, and Canyon support is stable

**Comment by protolambda** [comment]
We've one test that checks the halting behavior, and it was using the op-geth supported protocol version to determine the next protocol version to signal with. Even though op-geth does not have any Delta changes, I believe we do want it to signal that it is ready for Delta, to not warn/halt when it receives a Delta version signal. I'll make a op-geth PR to update that, and pull it into this PR.

**Review by clabby — APPROVED** [review]
Specs upgrade lgtm

**Review by ajsutton — APPROVED** [review]

**Pull request merged**